### PR TITLE
soft fail cluster tests for data gathering

### PR DESCRIPTION
**Pull request by davejrt**
These tests have been historically flaky, between infra and the tests themselves.

https://github.com/sourcegraph/sourcegraph/issues/25977
https://github.com/sourcegraph/sourcegraph/issues/31342

It's been challenging to gather enough accurate data so the intent here is to add a soft fail so we can get as many attempts as possible to see if there are any patterns we can observe. 
## Test plan

a test run here forcing the exit code to make soft fail is working
https://buildkite.com/sourcegraph/sourcegraph/builds/133725#5c966b24-2a16-4b30-a1ac-7628e4cf0312

### CODE REVIEW

File: dev/ci/integration/cluster/test.sh [hunk 1]
```diff
@@ -104,4 +104,4 @@ function e2e() {
 cluster_setup
 test_setup
 set +o pipefail
-e2e
+e2e || exit 1
```
Comment (bobheadxi) [inline dev/ci/integration/cluster/test.sh:107]: Maybe we should make this more obscure so other failures still come through as failures, i.e.

```suggestion
# special exit code to capture e2e failures
e2e || exit 123
```

this ensures we only soft fail on specifically the e2e tests failing

File: enterprise/dev/ci/internal/ci/operations.go [hunk 2]
```diff
@@ -593,7 +590,8 @@ func clusterQA(candidateTag string) operations.Operation {
 			bk.Env("TEST_USER_PASSWORD", "supersecurepassword"),
 			bk.Env("INCLUDE_ADMIN_ONBOARDING", "false"),
 			bk.Cmd("./dev/ci/integration/cluster/run.sh"),
-			bk.ArtifactPaths("./*.png", "./*.mp4", "./*.log"))
+			bk.ArtifactPaths("./*.png", "./*.mp4", "./*.log"),
+			bk.SoftFail(1, 2))
```
Comment (bobheadxi) [inline enterprise/dev/ci/internal/ci/operations.go:594]: based on the above comment:

```suggestion
			// Flakey test we are running to collect more data:
			// https://github.com/sourcegraph/sourcegraph/issues/31342
			bk.SoftFail(123))
```

**Review by bobheadxi — COMMENTED** [review]
good idea!

File: dev/ci/integration/cluster/test.sh (hunk 1 above)
Comment (davejrt) [reply dev/ci/integration/cluster/test.sh:107]: Nice, TY!

**Review by bobheadxi — APPROVED** [review]

**Pull request merged**